samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemCancellationListener.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2023 The Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.sakaiproject.tool.assessment.ui.listener.author;
+
+import javax.faces.application.FacesMessage;
+import javax.faces.context.FacesContext;
+import javax.faces.event.AbortProcessingException;
+import javax.faces.event.ActionEvent;
+import javax.faces.event.ActionListener;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.sakaiproject.samigo.util.SamigoConstants;
+import org.sakaiproject.tool.assessment.data.ifc.assessment.PublishedAssessmentIfc;
+import org.sakaiproject.tool.assessment.facade.ItemFacade;
+import org.sakaiproject.tool.assessment.services.PublishedItemService;
+import org.sakaiproject.tool.assessment.services.assessment.PublishedAssessmentService;
+import org.sakaiproject.tool.assessment.ui.bean.author.AssessmentBean;
+import org.sakaiproject.tool.assessment.ui.bean.evaluation.QuestionScoresBean;
+import org.sakaiproject.tool.assessment.ui.listener.evaluation.QuestionScoreListener;
+import org.sakaiproject.tool.assessment.ui.listener.util.ContextUtil;
+
+/**
+ * <p>Title: Samigo</p>
+ * <p>Description: Sakai Assessment Manager</p>
+ * <p>Organization: Sakai Project</p>
+ */
+@Slf4j
+public class ItemCancellationListener implements ActionListener {
+
+    public void processAction(ActionEvent ae) throws AbortProcessingException {
+        PublishedItemService publishedItemService = new PublishedItemService();
+        PublishedAssessmentService publishedAssessmentService = new PublishedAssessmentService();
+
+        String itemId = ContextUtil.lookupParam("itemId");
+        String cancellation = ContextUtil.lookupParam("cancellation");
+        String outcome = ContextUtil.lookupParam("outcome");
+        boolean regrade = Boolean.parseBoolean(ContextUtil.lookupParam("regrade"));
+
+        AssessmentBean assessmentBean = (AssessmentBean) ContextUtil.lookupBean("assessmentBean");
+        QuestionScoresBean questionScoresBean = (QuestionScoresBean) ContextUtil.lookupBean("questionScores");
+
+        log.debug("ItemCancel LISTENER.");
```

Comment:
Maybe better only one log.debug with all parameters... What do you think?

Reply:
Yeah, makes sense!